### PR TITLE
[obsolete] Allows to set retention params as env var and add database backup support

### PR DESCRIPTION
**Pull request by florentk**
resolve https://github.com/kiwix/borg-backup/issues/4 and https://github.com/kiwix/borg-backup/issues/3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,3 +13,15 @@ To backup files of `<barckupdir>` hourly with a new container :
...
 
+### Backup a database
+
+To backup a database we must define :
+
+- DB_TYPE : mysql or postgresql
+- DB_NAME
+- DB_USERNAME 
+- DB_PASSWORD
+
```
Comment (rgaudin) [inline README.md:24]: Pretty sure you'll also need the location/port of the server too

**Review by rgaudin — CHANGES_REQUESTED** [review]
I think we miss connection information to the databases. Has this been tested?
Keep in mind that this is gonna be a docker image companion to projects so even the default access won't work (`localhost:3306` for mysql)

**Comment by rgaudin** [comment]
But but I have concerns about the example docker-compose:

- should be in an `examples/` folder
- use regular networking:
  - no `network_mode` rule
  - no `ports` rule
  - use `- DB_HOSTNAME=db`
  - use `DB_NAME=all` for tests
  - use `build:` rule instead of image (or use public image)

Also, please create your PR from branches in this repo instead of your fork so that we can contribute to it directly.

**Comment by florentk** [comment]
> But but I have concerns about the example docker-compose:
> 
> * should be in an `examples/` folder

ok

> * use regular networking:
>   
>   * no `network_mode` rule
>   * use `- DB_HOSTNAME=db`

if only it was that simple, I try with this before use network_mode rule, but by default mysql database has not right to authorize connection from a other host that localhost ...

>   * no `ports` rule
>   * use `DB_NAME=all` for tests
>   * use `build:` rule instead of image (or use public image)
> 

ok

> Also, please create your PR from branches in this repo instead of your fork so that we can contribute to it directly.

ok

**Comment by rgaudin** [comment]
It is that simple. I've tested it.

```sh
docker exec -it borg-backup_backup_1 bash
root@0941467c13ed:~# mysql -h db -uroot -proot
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MySQL connection id is 36
Server version: 8.0.21 MySQL Community Server - GPL

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MySQL [(none)]> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| performance_schema |
| sys                |
| test_db            |
+--------------------+
5 rows in set (0.001 sec)

MySQL [(none)]>
```

**Comment by florentk** [comment]
> It is that simple. I've tested it.
> 
> ```shell
> docker exec -it borg-backup_backup_1 bash
> root@0941467c13ed:~# mysql -h db -uroot -proot
> Welcome to the MariaDB monitor.  Commands end with ; or \g.
> Your MySQL connection id is 36
> Server version: 8.0.21 MySQL Community Server - GPL
> 
> Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.
> 
> Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
> 
> MySQL [(none)]> show databases;
> +--------------------+
> | Database           |
> +--------------------+
> | information_schema |
> | mysql              |
> | performance_schema |
> | sys                |
> | test_db            |
> +--------------------+
> 5 rows in set (0.001 sec)
> 
> MySQL [(none)]>
> ```

so much the better, I don't know why yesterday it was complicated

**Pull request merged**